db/migrate/20200601212421_add_huid_to_household.rb

```diff
@@ -0,0 +1,6 @@
+class AddHuidToHousehold < ActiveRecord::Migration[6.0]
+  def change
+    add_column :households, :huid, :integer
```

Comment:
@MatthewSchultz I wonder if we should just implement this as a `SERIAL` instead of having the autoincrement logic in Rails. I don't _think_ we care about that from an application logic perspective - we just need a number we can use.

Reply:
👍  to Cory's comment. We should make that change.

Reply:
@bencalegari Matthew and I just chatted about this. The logic is that when a user starts an application, we create a household record for them. We do *not* want to create / autoincrement an ID at that point. Only when they get to the end at the signature page do we want to create the ID, which is why he set the logic like that.

We talked about whether that would happen in a transaction (both the `maximum` call and the save itself) and he said it does.

Reply:
Sadly, that's correct. Autoincriment won't work, since that fires when the record is created (that's why the id column uses it). With that in place, we'd overrun our address space quickly, since incomplete applications would still get HUIDs.

This was the best solution I could think of. Luckily, rails supports transactions. When I had it as a filter, it was automatically in a transaction (before_save/update are run inside transactions on rails). I expressly made it in a transaction in the controller to make sure it still is and prevent any collisions.